### PR TITLE
Upgrading Python 2.7 to 3.7

### LINKED ISSUE #18: Migrate to Python 3.7
We are going to drop support for Python 2.7 and migrate everything to Python 3.7.

### PR DESCRIPTION
**Pull request by neomatrix369**
Converted all .py files from python 2.x to 3.7. Also added instructions in the README

Fixes #18 

Pending, merging of this PR is dependent on 
- merging of PR https://github.com/julianghionoiu/tdl-client-test-wiremock/pull/1
- merging of PR https://github.com/julianghionoiu/tdl-client-test-broker/pull/6

And updating the respective git sub-modules (broker and wiremocks)

### CODE REVIEW

File: src/tdl/queue/implementation_runner_config.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-from tdl.audit.stdout_audit_stream import StdoutAuditStream
+from src.tdl.audit.stdout_audit_stream import StdoutAuditStream
```
Comment (julianghionoiu) [inline src/tdl/queue/implementation_runner_config.py]: We should not have "src" in front.
You should explore ways of loading the "src" folder with Python3

File: src/tdl/queue/processing_rules.py [hunk 2]
```diff
@@ -24,7 +24,7 @@ def get_response_for(self, request):
             result = processing_rule.user_implementation(*request.params)
             return ValidResponse(request.id, result)
         except Exception as e:
-            print(e.message)
+            print(e)
```
Comment (julianghionoiu) [inline src/tdl/queue/processing_rules.py]: If this going to print the entire stacktrace? If that is the case, then this is not desirable. We only want to provide a hint not the full stack.

File: src/tdl/runner/recording_system.py [hunk 3]
```diff
@@ -23,17 +23,18 @@ def is_recording_system_ok(self):
...
-            response = unirest.get("{}/status".format(RECORDING_SYSTEM_ENDPOINT))
+            response = requests.get("{}/status".format(RECORDING_SYSTEM_ENDPOINT))
 
-            if response.code == 200 and response.body.startswith("OK"):
+            response_body = response.text
+            if response.status_code == 200 and response_body.startswith("OK"):
                 return True
         except Exception as e:
-            print("Could not reach recording system: {}".format(str(e)))
+            print(("Could not reach recording system: {}".format(str(e))))
```
Comment (julianghionoiu) [inline src/tdl/runner/recording_system.py]: Why do we need multiple parentheses?

File: src/tdl/runner/recording_system.py [hunk 4]
```diff
@@ -23,17 +23,18 @@ def is_recording_system_ok(self):
...
 
         return False
 
     def notify_event(self, round_id, event_name):
-        print('Notify round "{}", event "{}"'.format(round_id, event_name))
+        print(('Notify round "{}", event "{}"'.format(round_id, event_name)))
```
Comment (julianghionoiu) [inline src/tdl/runner/recording_system.py]: same here

**Review by julianghionoiu — COMMENTED** [review]
We are almost there.
Only a few comments for sort out.

File: src/tdl/queue/implementation_runner_config.py (hunk 1 above)
Comment (julianghionoiu) [reply src/tdl/queue/implementation_runner_config.py]: This is the last thing to sort and we should be able to release the client.

**Comment by neomatrix369** [comment]
@julianghionoiu I ran the tests again and got this failing test:

```
Failing scenarios:
  test/features/queue/QueueRunner.feature:43  Handle multi-line request and response
```

It might explain why `tdl-runner-python` is also not working when I try to deploy.

**Comment by neomatrix369** [comment]
This was a red-herring btw - We get this issue when we run it in Python 2, instead of Python 3

**Comment by neomatrix369** [comment]
I have a small PR coming for this, although its not for any production related changes.

**Pull request merged**